websockets/websockets.go

```diff
@@ -147,7 +147,14 @@ func parseURL(c goja.ConstructorCall) (*url.URL, error) {
 func defineWebsocket(rt *goja.Runtime, w *webSocket) {
 	must(rt, w.obj.DefineDataProperty(
 		"addEventListener", rt.ToValue(w.addEventListener), goja.FLAG_FALSE, goja.FLAG_FALSE, goja.FLAG_TRUE))
-	// TODO add onmessage,onclose and so on.
+	must(rt, w.obj.DefineDataProperty(
+		"onMessage", rt.ToValue(w.generateEventListener(MESSAGE)), goja.FLAG_FALSE, goja.FLAG_FALSE, goja.FLAG_TRUE))
+	must(rt, w.obj.DefineDataProperty(
+		"onError", rt.ToValue(w.generateEventListener(ERROR)), goja.FLAG_FALSE, goja.FLAG_FALSE, goja.FLAG_TRUE))
+	must(rt, w.obj.DefineDataProperty(
+		"onOpen", rt.ToValue(w.generateEventListener(OPEN)), goja.FLAG_FALSE, goja.FLAG_FALSE, goja.FLAG_TRUE))
+	must(rt, w.obj.DefineDataProperty(
+		"onClose", rt.ToValue(w.generateEventListener(CLOSE)), goja.FLAG_FALSE, goja.FLAG_FALSE, goja.FLAG_TRUE))
```

Comment:
This isn't really immediately obvious from the specification but one of the main differences between `onerror` and `AddEventLisetner` with `error` is that the second can add multiple while the first one can only add *one* eventHandler. If you set it a second time it overwrites it.

I can't find the exact place where it was also talked on which one was fired first, but we likely can not use the addEventListener underlying mechanics and instead will have to add a field to the WebSocket struct per each of those.

Reply:
Okay, now I see that. Thanks! :+1: